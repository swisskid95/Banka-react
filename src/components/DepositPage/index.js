/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { Component } from 'react';
import SideNavPanel from '../commons/SideNavPanel';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import TopSection from '../commons/TopSection';
import PageTitle from '../commons/PageTitle';
import {
  fetchAccountDetails,
  performDeposit
} from '../../redux/actions/cashierStaffAction';

export class DepositPage extends Component {
  state = {
    accountNumber: '',
    remarks: '',
    amount: '',
    isLoading: false,
    showModal: false
  };

  handleInputChange = event => {
    event.preventDefault();
    const { target } = event;
    this.setState({ [target.name]: target.value });
  };

  handleConfirmDetails = async event => {
    const { fetchAccountDetails, user } = this.props;
    event.preventDefault();
    this.setState({ isLoading: true });
    fetchAccountDetails(user.token, this.state.accountNumber);
    this.setState({ isLoading: false });
    this.setState({ showModal: true });
  };

  handleModalClose = () => {
    this.setState({ showModal: false });
  };

  render() {
    const { user, customerDetails, performDeposit } = this.props;
    const { accountNumber, remarks, amount, isLoading, showModal } = this.state;

    const handlePerformDeposit = async () => {
      this.setState({ isLoading: true });
      await performDeposit(user.token, accountNumber, { remarks, amount });
      this.setState({ isLoading: false });
      this.setState({ showModal: false });
    };
    return (
      <main className="container">
        <SideNavPanel user={user} />
        <div className="main-wrapper">
          <TopSection />
          <PageTitle name={'deposit'} />

          <form
            onSubmit={this.handleConfirmDetails}
            className="entry--closer entry"
          >
            <input
              type="number"
              className="entry__input entry__input--bigger"
              placeholder="account number"
              name="accountNumber"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleInputChange}
            />
            <input
              type="number"
              className="entry__input entry-deposit__amount entry__input--bigger"
              placeholder="amount"
              name="amount"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleInputChange}
            />
            <textarea
              className="entry__input entry__input--remark entry__input--bigger"
              placeholder="remark"
              name="remarks"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleInputChange}
            ></textarea>
            <button
              disabled={isLoading}
              className="btn entry__btn entry__input--bigger modal__open"
            >
              deposit
            </button>
          </form>
        </div>

        <div
          onClick={this.handleModalClose}
          className={`modal ${showModal ? 'modal-open' : ''}`}
        >
          {!this.state.loading && Object.keys(customerDetails).length > 0 ? (
            <div href="#" className="transaction-full">
              <div className="transaction__details__title">
                <h1 className="content-title__text modal__title">
                  Confirm Deposit
                </h1>
              </div>
              <div className="transaction-full__main modal__main">
                <div className="transaction-full__type transaction-full__section">
                  <h4 className="transaction-full__title">Account Name</h4>
                  <p className="transaction-full__body">
                    {' '}
                    {`${customerDetails.accountDetails.owner.firstname} ${customerDetails.accountDetails.owner.lastname}`}
                  </p>
                </div>
                <div className="transaction-full__type transaction-full__section">
                  <h4 className="transaction-full__title">Account Number</h4>
                  <p className="transaction-full__body">
                    {customerDetails.accountDetails.accountnumber}
                  </p>
                </div>
                <div className="transaction-full__type transaction-full__section">
                  <h4 className="transaction-full__title">Amount</h4>
                  <p className="transaction-full__body">{amount}</p>
                </div>
                <div className="transaction-amount transaction-full__section">
                  <h4 className="transaction-full__title">remark</h4>
                  <p className="transaction-full__body">{remarks}</p>
                </div>
                <div className="transaction-full__section modal-response">
                  <button
                    className="btn modal__close action-center__btn btn-secondary"
                    onClick={this.handleModalClose}
                  >
                    cancel
                  </button>
                  <button
                    onClick={handlePerformDeposit}
                    className="btn action-center__btn"
                  >
                    confirm
                  </button>
                </div>
              </div>
            </div>
          ) : (
            'loading ...'
          )}
        </div>
      </main>
    );
  }
}

DepositPage.propTypes = {
  user: PropTypes.object,
  fetchAccountDetails: PropTypes.func,
  customerDetails: PropTypes.object,
  performDeposit: PropTypes.func
};

export const mapStateToProps = state => {
  return {
    user: state.users.userAuthDetails,
    customerDetails: state.customerDetails
  };
};

export const mapDispatchToProps = dispatch => ({
  fetchAccountDetails: (token, accountNumber) => {
    return dispatch(fetchAccountDetails(token, accountNumber));
  },
  performDeposit: (token, accountNumber, depositData) => {
    return dispatch(performDeposit(token, accountNumber, depositData));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DepositPage);
