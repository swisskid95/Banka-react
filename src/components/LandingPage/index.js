import React from 'react';
import { Link } from 'react-router-dom';

const LandingPage = () => {
  return (
    <main className="container__entry">
      <div className="landing__page">
        <div className="section-wrapper">
          <section className="main-section">
            <h1 className="logo--big">Banka</h1>
            <h2 className="first-text">Banking made easier</h2>
            <p className="second-text">
              Banka helps you track all transaction on your account(s)
            </p>
            <Link to="/sign-up" className="cta">
              SIGN-UP{' '}
            </Link>
          </section>
          <section className="infograph-section">
            <img
              className="monitor"
              src="../../src/assets/images/infograph.svg"
              alt="infograph"
            />
          </section>
        </div>
      </div>
    </main>
  );
};

export default LandingPage;
