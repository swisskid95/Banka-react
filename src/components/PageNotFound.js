import React from 'react';
import TopSection from '../components/commons/TopSection';

const PageNotFound = () => (
  <main className="container">
    <div className="main-wrapper">
      <TopSection />
      <h1 className="page-not-found">Oops! Page not found.</h1>
    </div>
  </main>
);

export default PageNotFound;
