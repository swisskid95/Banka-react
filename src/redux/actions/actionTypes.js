export const LOGIN_USER = 'LOGIN_USER';
export const LOGIN_USER_ERROR = 'LOGIN_USER_ERROR';
export const SIGN_UP_USER = 'SIGN_UP_USER';
export const SIGN_UP_USER_ERROR = 'SIGN_UP_USER_ERROR';
export const ADD_ACCOUNT_DETAILS = 'ADD_ACCOUNT_DETAILS';
export const ADD_ALL_ACCOUNTS = 'ADD_ALL_ACCOUNTS';
export const ADD_ALL_ACCOUNTS_ERROR = 'ADD_ALL_ACCOUNTS_ERROR';
export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const GET_USER_DETAILS = 'GET_USER_DETAILS';
