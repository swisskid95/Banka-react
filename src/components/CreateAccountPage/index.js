import React, { Component } from 'react';
import SideNavPanel from '../commons/SideNavPanel';
import TopSection from '../commons/TopSection';
import { connect } from 'react-redux';
import { createAccountRequest } from '../../redux/actions/createAccountActions';
import PropTypes from 'prop-types';
export class CreateAccountPage extends Component {
  state = {
    type: '',
    accountPurpose: '',
    isLoading: false
  };

  handleInputChange = event => {
    const { target } = event;
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    const user = JSON.parse(localStorage.getItem('user'));
    this.setState({ isLoading: true });
    await this.props.createAccountRequest(user.token, this.state);
    this.setState({ isLoading: false });
  };

  render() {
    const { isLoading } = this.state;
    return (
      <main className="container">
        <SideNavPanel />
        <div className="main-wrapper">
          <TopSection />
          <div className="content-title">
            <h1 className="content-title__text">create account</h1>
          </div>
          <form onSubmit={this.handleSubmit} className="entry--closer entry">
            <select
              name="type"
              onBlur={this.handleInputChange}
              className="entry__input--account-type entry__input entry__input--bigger"
            >
              <option value="current">current</option>
              <option value="savings">savings</option>
            </select>
            <input
              type="text"
              onChange={this.handleInputChange}
              className="entry__input entry__input--bigger"
              name="accountPurpose"
              placeholder="account purpose"
            />
            <button
              className={`btn entry__btn entry__input--bigger ${
                isLoading ? 'btn-secondary' : ''
              }`}
            >
              {isLoading ? 'Loading...' : 'create account'}
            </button>
          </form>
        </div>
      </main>
    );
  }
}

CreateAccountPage.propTypes = {
  createAccountRequest: PropTypes.func.isRequired
};

export const mapDispatchToProps = dispatch => {
  return {
    createAccountRequest: async (token, data) =>
      dispatch(await createAccountRequest(token, data))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(CreateAccountPage);
