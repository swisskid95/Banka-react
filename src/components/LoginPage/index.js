/* eslint-disable no-console */
import React, { Component } from 'react';
import EntrySideBar from '../commons/EntrySideBar';
import TopSection from '../commons/TopSection';
import PageTitle from '../commons/PageTitle';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loginUserRequest } from '../../redux/actions/loginActions';

export class LoginPage extends Component {
  state = {
    email: '',
    password: '',
    isLoading: false
  };

  handleInputChange = event => {
    event.preventDefault();
    const { target } = event;

    this.setState({ [target.name]: target.value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({ isLoading: true });
    await this.props.loginUserRequest(this.state, this.props.history);
    this.setState({ isLoading: false });
  };

  render() {
    const { isLoading } = this.state;
    return (
      <main className="container__entry">
        <EntrySideBar
          welcomeText="HELLO, Friend"
          mainText="Not yet a member, click on the button below to Join in the benefits."
          buttonText="sign-up"
        />
        <TopSection />
        <PageTitle name={'login'} />
        <form className="entry" onSubmit={this.handleSubmit}>
          <div className="field">
            <input
              className="entry__input"
              type="email"
              placeholder="Email"
              name="email"
              required
              onChange={this.handleInputChange}
            />
            <span className="helper-text"></span>
          </div>
          <div className="field">
            <input
              className="entry__input"
              type="password"
              placeholder="Password"
              name="password"
              required
              onChange={this.handleInputChange}
            />
            <span className="helper-text"></span>
          </div>
          <button
            className={`entry__btn btn ${isLoading ? 'btn-secondary' : ''}`}
          >
            {isLoading ? 'Loading...' : 'Login'}
          </button>
        </form>
        <div className="switch">
          <p>
            Not yet a member?
            <Link to="/sign-up" className="switch__link">
              sign-up
            </Link>
          </p>
        </div>
      </main>
    );
  }
}

LoginPage.propTypes = {
  loginUserRequest: PropTypes.func,
  history: PropTypes.any
};

export const mapDispatchToProps = dispatch => {
  return {
    loginUserRequest: async (loginCredentials, history) =>
      dispatch(await loginUserRequest(loginCredentials, history))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(LoginPage);
