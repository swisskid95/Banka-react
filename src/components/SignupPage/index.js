import React, { Component } from 'react';
import EntrySideBar from '../commons/EntrySideBar';
import { Link } from 'react-router-dom';
import TopSection from '../commons/TopSection';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signUpUserRequest } from '../../redux/actions/signupActions';
import * as validate from '../../utils/validation';

export class SignupPage extends Component {
  state = {
    firstName: '',
    lastName: '',
    phoneNumber: '',
    password: '',
    isLoading: false,
    disableSubmit: true
  };

  handleValidation = event => {
    const { target } = event;

    let unDisableSubmit;

    unDisableSubmit = !validate.isEmpty(target);

    if (target.name === 'fullName') {
      unDisableSubmit = !validate.isValidFullName(target);
    }
    if (target.name === 'password') {
      unDisableSubmit = !validate.meetSpecifiedLength(target, 7, 100);
    }
    if (target.name === 'phoneNumber') {
      unDisableSubmit = !validate.matchNigeriaPhoneNumber(target);
    }
    if (target.name === 'email') {
      unDisableSubmit = !validate.isValidEmailAddress(target);
    }

    this.setState({ disableSubmit: unDisableSubmit });
  };

  handleInputChange = event => {
    event.preventDefault();

    this.handleValidation(event);

    const { target } = event;

    if (target.name === 'fullName') {
      const [firstName, lastName] = target.value.split(' ');
      if (firstName && lastName) {
        this.setState({ firstName, lastName });
      }
    } else {
      this.setState({ [target.name]: target.value });
    }
  };

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({ isLoading: true });
    await this.props.signUpUserRequest(this.state, this.props.history);
    this.setState({ isLoading: false });
  };

  render() {
    const { disableSubmit, isLoading } = this.state;
    return (
      <main className="container__entry">
        <EntrySideBar
          welcomeText="Welcome back"
          mainText="Already a member, click on the button below to Sign in to continue
                your benefits."
          buttonText="login"
        />
        <TopSection />
        <div className="content-title">
          <h1 className="content-title__text">sign up</h1>
        </div>
        <form onSubmit={this.handleSubmit} className="entry">
          <div className="field">
            <input
              className="entry__input"
              type="text"
              placeholder="Full Name"
              id="fullName"
              name="fullName"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleValidation}
            />
            <span className="helper-text"></span>
          </div>
          <div className="field">
            <input
              className="entry__input"
              type="tel"
              placeholder="Phone Number"
              name="phoneNumber"
              id="phoneNumber"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleValidation}
            />
            <span className="helper-text"></span>
          </div>
          <div className="field">
            <input
              className="entry__input"
              type="email"
              placeholder="Email"
              name="email"
              id="email"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleValidation}
            />
            <span className="helper-text"></span>
          </div>
          <div className="field">
            <input
              className="entry__input"
              type="password"
              placeholder="Password"
              name="password"
              id="password"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleValidation}
            />
            <span className="helper-text"></span>
          </div>
          <button
            disabled={disableSubmit || isLoading}
            className={`entry__btn btn ${
              isLoading || disableSubmit ? 'btn-secondary' : ''
            }`}
            id="submit"
          >
            {isLoading ? 'Loading...' : 'Sign up'}
          </button>
        </form>
        <div className="switch">
          <p>
            Already a member?
            <Link to="/login" className="switch__link">
              Login
            </Link>
          </p>
        </div>
      </main>
    );
  }
}

SignupPage.propTypes = {
  signUpUserRequest: PropTypes.func,
  history: PropTypes.any
};

export const mapDispatchToProps = dispatch => {
  return {
    signUpUserRequest: async (signupCredentials, history) =>
      dispatch(await signUpUserRequest(signupCredentials, history))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(SignupPage);
