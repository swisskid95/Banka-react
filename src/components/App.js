import React from 'react';
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import PageNotFound from './PageNotFound';
import LandingPage from './LandingPage';
import SignupPage from './SignupPage';
import LoginPage from './LoginPage';
import SummaryPage from './SummaryPage';
import CreateAccountPage from './CreateAccountPage';
import { ToastContainer } from 'react-toastify';
import DepositPage from './DepositPage';
import WithdrawalPage from './WithdrawalPage';
import PrivateRoute from './PrivateRoute';

class App extends React.Component {
  render() {
    return (
      <Router>
        <ToastContainer />
        <Switch>
          <Route exact path="/" component={LandingPage} />
          <Route path="/sign-up" component={SignupPage} />
          <Route path="/login" component={LoginPage} />
          <PrivateRoute path="/summary" component={SummaryPage} />
          <PrivateRoute path="/create-account" component={CreateAccountPage} />
          <PrivateRoute path="/deposit" component={DepositPage} />
          <PrivateRoute path="/withdraw" component={WithdrawalPage} />
          <Route component={PageNotFound} />
        </Switch>
      </Router>
    );
  }
}

export default hot(module)(App);
