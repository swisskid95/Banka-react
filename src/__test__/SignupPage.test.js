/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { SignupPage, mapDispatchToProps } from '../components/SignupPage';
import { BrowserRouter } from 'react-router-dom';
import {
  signUpUserRequest,
  signUpUserError
} from '../redux/actions/signupActions';
import thunk from 'redux-thunk';
import mockAxios from 'axios';
import configureMockStore from 'redux-mock-store';
import { SIGN_UP_USER_ERROR, SIGN_UP_USER } from '../redux/actions/actionTypes';
import signupReducer from '../redux/reducers/authReducer';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const renderSignupPage = args => {
  const defaultProps = {
    signUpUserRequest: jest.fn(),
    history: {},
    match: {}
  };
  const props = { ...defaultProps, ...args };
  return mount(
    <BrowserRouter>
      <SignupPage {...props} />
    </BrowserRouter>
  );
};

describe('Login in Components Tests', () => {
  it('renders Sign page In template', () => {
    const wrapper = renderSignupPage();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
  });

  it(`Simulates an onchange event on form email input`, () => {
    const wrapper = renderSignupPage();
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { currentTarget: { value: 'fola@gmail.com' } });
  });

  it(`Simulates an onchange event on form password input`, () => {
    const wrapper = renderSignupPage();
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { currentTarget: { value: 'password' } });
  });
  it('Simulates a form submit event', () => {
    const wrapper = renderSignupPage();
    wrapper.find('form').simulate('submit');
  });

  it('should dispatch signup request action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).signUpUserRequest();
  });
});

describe('sign Async action Tests', () => {
  const userCredentials = {
    email: 'someuser@gmail.com',
    password: 'randompassword',
    firstName: 'some',
    lastName: 'user'
  };

  let store;
  beforeEach(() => {
    store = mockStore({});
    jest.resetAllMocks();
  });
  afterEach(() => {
    store.clearActions();
  });

  it('Should Trigger the SIGN_UP_USER dispatch function', async () => {
    const mockData = {
      data: {
        firstName: 'some',
        lastName: 'user',
        email: 'someuser@gmail.com',
        token:
          'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MjMsI…c4OX0.DfJClt28K3qIgJJWp9FMypDO0tKD1NoiNtO9YvEOvsQ'
      }
    };

    mockAxios.post.mockResolvedValue({
      status: 200,
      data: mockData
    });
    const historyObject = {
      push: jest.fn()
    };
    await store.dispatch(signUpUserRequest(userCredentials, historyObject));

    expect(store.getActions()).toEqual([]);
  });

  it('Should Trigger the LOGIN_USER_ERROR dispatch function', async () => {
    const mockData = {
      data: {
        data: {
          message: 'invalid email/password'
        }
      }
    };

    mockAxios.post.mockResolvedValue({
      response: mockData
    });

    const expectedActions = [
      { type: SIGN_UP_USER_ERROR, payload: mockData.data.data.message }
    ];
    const { message } = mockData.data.data;
    store.dispatch(signUpUserError(message));
    expect(store.getActions()).toEqual(expectedActions);
  });
});

describe('signup Reducer Tests', () => {
  it('Should return default state', () => {
    const newState = signupReducer(undefined, {});
    expect(newState).toEqual({});
  });
  it('Should return a new state if it recieves a signup action type', () => {
    const user = {
      firstName: 'fola',
      lastName: 'abass',
      email: 'abass@gmail.com',
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MjMsI…c4OX0.DfJClt28K3qIgJJWp9FMypDO0tKD1NoiNtO9YvEOvsQ'
    };
    const newState = signupReducer(undefined, {
      type: SIGN_UP_USER,
      payload: user
    });
    expect(newState).toEqual({ user });
  });

  it('Should return a new state if it recieves a log in error action type', () => {
    const error = {
      message: 'invalid email/password'
    };
    const newState = signupReducer(undefined, {
      type: SIGN_UP_USER_ERROR,
      payload: error
    });
    expect(newState).toEqual({ user: error });
  });
});
