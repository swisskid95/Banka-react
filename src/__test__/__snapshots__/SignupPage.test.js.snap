// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up in Components Tests renders Sign page In template 1`] = `
<main
  className="container__entry"
>
  <aside
    className="sidebar"
  >
    <div
      className="sidebar__close"
    >
      Ã—
    </div>
    <h1
      className="sidebar__title"
    >
      Welcome back
    </h1>
    <p
      className="sidebar__text"
    >
      Already a member, click on the button below to Sign in to continue your benefits.
    </p>
    <a
      className="sidebar__btn"
      href="/login"
      onClick={[Function]}
    >
      login
    </a>
  </aside>
  <div
    className="top-section"
  >
    <img
      alt="Sidebar open button"
      className="sidebar__open"
      src="/src/assets/images/menu.svg"
    />
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      <h1
        className="logo"
      >
        Banka
      </h1>
    </a>
  </div>
  <div
    className="content-title"
  >
    <h1
      className="content-title__text"
    >
      sign up
    </h1>
  </div>
  <form
    className="entry"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <input
        className="entry__input"
        id="fullName"
        name="fullName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Full Name"
        required={true}
        type="text"
      />
      <span
        className="helper-text"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="entry__input"
        id="phoneNumber"
        name="phoneNumber"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Phone Number"
        required={true}
        type="tel"
      />
      <span
        className="helper-text"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="entry__input"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
      />
      <span
        className="helper-text"
      />
    </div>
    <div
      className="field"
    >
      <input
        className="entry__input"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <span
        className="helper-text"
      />
    </div>
    <button
      className="entry__btn btn btn-secondary"
      disabled={true}
      id="submit"
    >
      Sign up
    </button>
  </form>
  <div
    className="switch"
  >
    <p>
      Already a member?
      <a
        className="switch__link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </p>
  </div>
</main>
`;
